From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Martijn Muijsers <martijnmuijsers@live.nl>
Date: Tue, 29 Nov 2022 01:17:03 +0100
Subject: [PATCH] Hide irrelevant compilation warnings

Hides some compilation warnings.
This is done because Paper code contains many instances that trigger these warnings, leading to them filling the terminal every time it is compiled, and making it harder to find errors among the wall of text.

This patch has an analogous patch in `fiddle-api` with the same name.

License: GPL-3.0 (https://www.gnu.org/licenses/gpl-3.0.html)
Fiddle - https://fiddlemc.org

This patch is based on the following patch:
"Hide irrelevant compilation warnings"
By: Martijn Muijsers <martijnmuijsers@live.nl>
As part of: Gale (https://github.com/GaleMC/Gale)
Licensed under: GPL-3.0 (https://www.gnu.org/licenses/gpl-3.0.html)

* Gale description *

License: GPL-3.0 (https://www.gnu.org/licenses/gpl-3.0.html)
Gale - https://galemc.org

diff --git a/build.gradle.kts b/build.gradle.kts
index 09bf8edf8bc7681f6eefc89fa429e734fb7ad30b..4073145e01b34ec31ce16e5a75f4634a6c192e99 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -68,6 +68,16 @@ dependencies {
 }
 
 val craftbukkitPackageVersion = "1_20_R1" // Paper
+
+// Fiddle start - Gale - hide irrelevant compilation warnings
+tasks.withType<JavaCompile> {
+    val compilerArgs = options.compilerArgs
+    compilerArgs.add("-Xlint:-module")
+    compilerArgs.add("-Xlint:-removal")
+    compilerArgs.add("-Xlint:-dep-ann")
+}
+// Fiddle end - Gale - hide irrelevant compilation warnings
+
 tasks.jar {
     archiveClassifier.set("dev")
 
